src/Schema/AST/ASTCache.php

```diff
@@ -59,18 +59,21 @@ public function __construct(ConfigRepository $config)
         $this->enable = $cacheConfig['enable'];
 
         $version = $cacheConfig['version'] ?? 1;
-        if (! in_array($version, [1, 2])) {
-            throw new UnknownCacheVersionException($version);
-        }
-        $this->version = $version;
 
-        // Version 1
-        $this->store = $cacheConfig['store'] ?? null;
-        $this->key = $cacheConfig['key'];
-        $this->ttl = $cacheConfig['ttl'];
+        switch ($version) {
```

Comment:
if we think this is gonna be more and more of a thing, with new caching versions etc. we could consider making a driver based system, so all of the configs needed depending on which version is in it's own class

Reply:
Yeah this might even make more sense in the future!